src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -188,6 +195,101 @@ public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerial
             return default;
         }
 
+        private static bool IsNumericString(ref Utf8JsonReader reader)
+        {
+            ReadOnlySpan<byte> span = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+
+            // Numeric value with Leading and trailing whitespaces should be treated as valid based on behavior of Enum.TryParse()
+#if NETCOREAPP
+            span = span.Trim((byte)' ');
+#else
+            int leftIdx = 0;
+            for (; leftIdx < span.Length; leftIdx++)
+            {
+                if (span[leftIdx] != ' ')
+                {
+                    break;
+                }
+            }
+
+            int rightIdx = span.Length - 1;
+            for (; rightIdx > leftIdx; rightIdx--)
+            {
+                if (span[rightIdx] != ' ')
+                {
+                    break;
+                }
+            }
+
+            span = span.Slice(leftIdx, rightIdx - leftIdx + 1);
+#endif
+            if (span.Length > 1 && span[0] == '+')
```

Comment:
I think much of this logic could be replaced with a simple regex.

Reply:
Use regex now.